spyder/widgets/explorer.py

```diff
@@ -1201,16 +1202,9 @@ def __init__(self, parent=None, name_filters=['*.py', '*.pyw'],
             widget.setIconSize(QSize(16, 16))
 
         # Layouts
-        blayout = QHBoxLayout()
-        blayout.addWidget(button_previous)
-        blayout.addWidget(button_next)
-        blayout.addWidget(button_parent)
-        blayout.addStretch()
-        blayout.addWidget(self.button_menu)
-
-        layout = QVBoxLayout()
-        layout.addLayout(blayout)
-        layout.addWidget(self.treewidget)
+        controls = [button_previous, button_next, button_parent, 'stretch',

```

Comment:
I see the list `controls` as a recipe telling the function how the layout should look like.

I'm not dogmatic about `'stetch'`. It's basically a convenience, because you'll often need it in the layout and would have to manually create the horizontal layouts if you don't have something like this. So it saves work.

If you're using `None` as a separator elsewhere we might go with that. Even though I would strongly go for `'stetch'`, when building from scratch. It's more explicit than `None` and more compact than adding an additional comment. Maybe even better would be `'add_stretch'`. Also, it's extendable in the future, if you want a fixed stretch (`'add_stretch(10)'`).